component/config.go

```diff
@@ -33,22 +33,15 @@ type ConfigValidator interface {
 // ValidateConfig validates a config, by doing this:
 //   - Call Validate on the config itself if the config implements ConfigValidator.
 func ValidateConfig(cfg Config) error {
-	var err error
-	errs := validate(reflect.ValueOf(cfg))
-
-	for _, suberr := range errs {
-		err = errors.Join(err, suberr.toError())
-	}
-
-	return err
+	return errors.Join(validate(reflect.ValueOf(cfg)))
```

Comment:
Ah, damn you would need to change the return type of `validate` or cast items one by one :cry:

Reply:
Sorry, I hadn't thought about this. I still like using the standard Go interfaces more, but I see less value in my suggestion now

Reply:
I can't find a way to pull this off. The spread operator doesn't work since the type system still considers it a slice and not separate arguments. Changing the return type of `validate` will require a ton of casting since we use the `pathError` type directly in recursive calls to `validate`.

I liked consolidating this onto one line, any other ideas for how to solve this?

Reply:
We could reapply the loop with what's here, though part of the reason I had the `toError` method create new errors instead of having `pathError` implement `error` was to keep the `pathError` type internal and remove it from the error chain (not that it really hurts anything).

Reply:
I'm fine sticking with the standard Go interfaces even if the `pathError` type ends up in the chain. I've reintroduced the loop in `ValidateConfig`.

Reply:
I have not been able to find a way, let's keep it like this if you are okay with it